components/cmake/common_setup.cmake

```diff
@@ -113,69 +107,12 @@ else()
   endif()
 endif()
 
-# JGF TODO: replace with findnetcdf
-if (NETCDF_C_PATH)
-  if (NOT NETCDF_FORTRAN_PATH)
-    message(FATAL_ERROR "NETCDF_C_PATH specified without NETCDF_FORTRAN_PATH")
-  endif()
-  set(NETCDF_SEPARATE TRUE)
-  if (NOT INC_NETCDF_C)
-    set(INC_NETCDF_C ${NETCDF_C_PATH}/include)
-  endif()
-  if (NOT INC_NETCDF_FORTRAN)
-    set(INC_NETCDF_FORTRAN ${NETCDF_FORTRAN_PATH}/include)
-  endif()
-  if (NOT LIB_NETCDF_C)
-    if (EXISTS ${NETCDF_C_PATH}/lib)
-      set(LIB_NETCDF_C ${NETCDF_C_PATH}/lib)
-    elseif (EXISTS ${NETCDF_C_PATH}/lib64)
-      set(LIB_NETCDF_C ${NETCDF_C_PATH}/lib64)
-    else()
-      message(FATAL_ERROR "NETCDF_C_PATH does not contain a lib or lib64 directory")
-    endif()
-  endif()
-  if (NOT LIB_NETCDF_FORTRAN)
-    if(EXISTS ${NETCDF_FORTRAN_PATH}/lib)
-      set(LIB_NETCDF_FORTRAN ${NETCDF_FORTRAN_PATH}/lib)
-    elseif(EXISTS ${NETCDF_FORTRAN_PATH}/lib64)
-      set(LIB_NETCDF_FORTRAN ${NETCDF_FORTRAN_PATH}/lib64)
-    else()
-      message(FATAL_ERROR "NETCDF_FORTRAN_PATH does not contain a lib or lib64 directory")
-    endif()
-  endif()
-elseif (NETCDF_FORTRAN_PATH)
-  message(FATAL_ERROR "NETCDF_FORTRAN_PATH specified without NETCDF_C_PATH")
-elseif (NETCDF_PATH)
-  set(NETCDF_SEPARATE FALSE)
-  if (NOT INC_NETCDF)
-    set(INC_NETCDF ${NETCDF_PATH}/include)
-  endif()
-  if (NOT LIB_NETCDF)
-    if (EXISTS ${NETCDF_PATH}/lib)
-      set(LIB_NETCDF ${NETCDF_PATH}/lib)
-    elseif(EXISTS ${NETCDF_PATH}/lib64)
-      set(LIB_NETCDF ${NETCDF_PATH}/lib64)
-    else()
-      message(FATAL_ERROR "NETCDF_PATH does not contain a lib or lib64 directory")
-    endif()
-  endif()
-else()
-  message(FATAL_ERROR "NETCDF not found: Define NETCDF_PATH or NETCDF_C_PATH and NETCDF_FORTRAN_PATH in config_machines.xml or config_compilers.xml")
-endif()
-
-if (MPILIB STREQUAL mpi-serial)
-  if (PNETCDF_PATH)
-    unset(PNETCDF_PATH)
-  endif()
-else()
-  if (PNETCDF_PATH)
-    if (NOT INC_PNETCDF)
-      set(INC_PNETCDF ${PNETCDF_PATH}/include)
-    endif()
-    if (NOT LIB_PNETCDF)
-      set(LIB_PNETCDF ${PNETCDF_PATH}/lib)
-    endif()
-  endif()
+# Have to do this here for now because the macros must be loaded in order for
+# the netcdf cmake variables to be set. Once scorpio has a good package-config,
+# this won't be necessary
+if (NOT TARGET netcdf)
```

Comment:
I don't think we need this if guard. subsequent calls to `find_package` will do nothing if the package was already found...

Reply:
I think that depends on the implementation, but I'm not sure. I wanted to keep the findX modules as simple as possible, so they don't set or check XX_FOUND

Reply:
I think CMake states it clearly
```
Once one of the calls succeeds the result variable will be set and stored in the cache so that no call will search again.
```

Reply:
That could be for find_package when it does things the "normal" way, IE loading the package-config.cmake. When you are rolling your own find module, I don't think this is true. The example FindXXX.cmake guidance in the documentation has all that stuff explicit.

Reply:
Ok, no big deal. I think the logic for skipping second calls works regardless of search mode (provided that FindXXX.cmake sets `<PackageName>_FOUND=TRUE`). But I am not entirely sure, so let's not stop progress based on this detail.